### PR TITLE
WTCET-18

### PR DESCRIPTION
**Pull request by PaddyCello**
Hi Puneet! Here is my PR for WTCET-18 - please let me know your thoughts.

One thing that I will mention is that I am using a TDD approach to **everything** - any instance variables etc that are currently missing will be added in subsequent tickets, as and when they are required according to ticket specifications. If this approach is risky, do please flag it!

### CODE REVIEW

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java [hunk 1]
```diff
@@ -1,10 +1,136 @@
...
+	
+	private List<Student> allStudents = new ArrayList<Student>();
+	
+	//Method for creating new Student from user input
+	public int[] createStudent(InputStream inputStream) throws NumberFormatException {
+		
+		//Create new Scanner and take user input
+		Scanner scanner = new Scanner(inputStream);
+		
+		System.out.println("How many students are you adding?");
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: No System Out statement

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java [hunk 2]
```diff
@@ -1,10 +1,136 @@
...
+		//Return early if user enters number lower than one
+		if (numOfTimes < 1) {
+			scanner.close();
+			
+			logger.log(Level.WARNING, "Number of Students too low.");
+			return null;
+		}
+		
+		//Create new ArrayList to hold IDs of successfully created Students
+		ArrayList<Integer> studentIDs = new ArrayList<Integer>();
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: All the variables should be declared at the start of the method. It should be defined whereever necessary.
ArrayList<Integer> studentIDs = null; (At the start of the method)
studentIDs = new ArrayList<Integer>(); should be defined wherever necessary.

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java [hunk 3]
```diff
@@ -1,10 +1,136 @@
...
+import java.util.logging.*;
 
+public class Admin {
+	
+	Logger logger = Logger.getLogger("com.pjohnson_wtc.student_db_admin.admin");
+	
+	private List<Student> allStudents = new ArrayList<Student>();
+	
+	//Method for creating new Student from user input
+	public int[] createStudent(InputStream inputStream) throws NumberFormatException {
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: You are not throwing NumberFormatException???

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java (hunk 1 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: Hi Puneet,

If I wasn't using a Scanner to communicate with the user via the console, I wouldn't use System.out; however, please do let me know if there's a better alternative when using Scanners.

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java [hunk 4]
```diff
@@ -1,10 +1,136 @@
...
+		return studentIDPrimitive;
+		
+	}
+	
+	//Check for valid year and create Student from input
+	private Student checkYearAndCreateStudent(String stringYear, ArrayList<Integer> studentIDs, String firstName, String lastName) {
+		
+		int year = Integer.parseInt(stringYear);
+		
+		//Early return if year is not valid
```
Comment (puneetg1) [inline src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java:92]: Validation should be done separately and logged properly.

**Review by puneetg1 — CHANGES_REQUESTED** [review]

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java (hunk 2 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: Of course - will change :)

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java (hunk 3 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: Got it - will remove

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java (hunk 4 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java:92]: Fab, will do

File: src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java (hunk 1 above)
Comment (PaddyCello) [reply src/main/java/com/pjohnson_wtc/student_db_admin/Admin.java]: Logger works - woohoo! Will change them over

**Comment by PaddyCello** [comment]
Hi Puneet! I have made all of the changes you have requested for this PR - plus I added .classpath to .gitignore, as this is a general point rather than one specific to WTCET-19. Let me know if all is good.

**Review by puneetg1 — APPROVED** [review]

**Pull request merged**